src/Functions/FunctionsHashing.h

```diff
@@ -274,6 +274,54 @@ struct SipHash64Impl
     static constexpr bool use_int_hash_for_pods = false;
 };
 
+struct SipHash64KeyedImpl
+{
+    static constexpr auto name = "sipHash64Keyed";
+    using ReturnType = UInt64;
+    struct KeyType
+    {
+        UInt64 key0 = 0;
+        UInt64 key1 = 0;
+    };
+
+    static KeyType parseKey(const ColumnWithTypeAndName & key)
```

Comment:
I am fine with a static function in SipHash64KeyedImpl. Since SipHash128 also calls it (l. 350), we could have placed the code also into a "detail"/"impl" namespace (*), e.g. 

```cpp
namespace impl
{

static KeyType parseSipHashKey(const ColumnWithTypeAndName & key)
{
    ...
}

}
```

EDIT: Same could be done for the `KeyType` since that is shared with sipHash128 (l. 348)

```cpp
namespace impl
{

struct SipHashKey
{
...
};

}
```

(*) Normally, anonymous namespaces are used for that but they are not recommended in headers.